Plan/src/test/java/com/djrapitops/plan/system/database/databases/H2Test.java

```diff
@@ -0,0 +1,1060 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.djrapitops.plan.system.database.databases;
+
+import com.djrapitops.plan.DaggerPlanBukkitComponent;
+import com.djrapitops.plan.PlanBukkitComponent;
+import com.djrapitops.plan.api.exceptions.database.DBInitException;
+import com.djrapitops.plan.data.WebUser;
+import com.djrapitops.plan.data.container.*;
+import com.djrapitops.plan.data.store.Key;
+import com.djrapitops.plan.data.store.containers.AnalysisContainer;
+import com.djrapitops.plan.data.store.containers.NetworkContainer;
+import com.djrapitops.plan.data.store.containers.PlayerContainer;
+import com.djrapitops.plan.data.store.containers.ServerContainer;
+import com.djrapitops.plan.data.store.keys.*;
+import com.djrapitops.plan.data.store.objects.Nickname;
+import com.djrapitops.plan.data.time.GMTimes;
+import com.djrapitops.plan.data.time.WorldTimes;
+import com.djrapitops.plan.system.PlanSystem;
+import com.djrapitops.plan.system.database.DBSystem;
+import com.djrapitops.plan.system.database.databases.sql.H2DB;
+import com.djrapitops.plan.system.database.databases.sql.SQLDB;
+import com.djrapitops.plan.system.database.databases.sql.tables.*;
+import com.djrapitops.plan.system.info.server.Server;
+import com.djrapitops.plan.system.settings.Settings;
+import com.djrapitops.plan.utilities.Base64Util;
+import com.djrapitops.plan.utilities.SHA256Hash;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import utilities.OptionalAssert;
+import utilities.RandomData;
+import utilities.TestConstants;
+import utilities.mocks.PlanBukkitMocker;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test for the H2 database
+ *
+ * @author Rsl1122, Fuzzlemann
+ * @see SQLiteTest
+ * @since 4.5.1
+ */
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class H2Test {
```

Comment:
Could this test class be abstracted - so that both SQLiteTest and H2Test extend the same abstract test class?
This would reduce the amount of duplicate code in the database testing classes.

Reply:
Some tests were removed because of being not compatible (the reusing of the primary key)

Reply:
Done in 0a9cf9c241f691947cbfa0f781a67efd924c05a6